File: java/69Sqrtx.java
class Solution {
    public int mySqrt(int x) {
        if(x==0 || x == 1) return x;
        int left = 1;
        int right = x/2;
        while(left+1 < right) {
            int mid = left+(right-left)/2;
            System.out.println(mid);
            if(mid == x/mid) {
                return mid;   
            }else if(mid < x/mid) {
                left = mid;
            }else {
                right = mid-1;
            }
        }
        return right <= x/right ? right:left;
    }
}

File: python3/69_Sqrt_x.py
class Solution:
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x<1:
            return 0

        l = 0
        r = x+1
        while r-l>1:
            temp = (r-l)//2+l
            if temp*temp == x:
                return temp
            elif temp*temp <x:
                if (temp+1)*(temp+1)>x:
                    return temp
                l = temp
            else:
                
                r = temp
        
        if l*l>x:
            return l-1
        elif r*r>x:
            return r-1
            